dracut/repartition/endless-repartition.sh

```diff
@@ -121,14 +121,31 @@ if [ "$pt_label" = "dos" ]; then
 fi
 
 # check the last partition on the disk
-lastpart=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
+lastpartno=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
+
+# If we detect a recovery partition at the end of the disk, we'll be careful
+# not to destroy it.
+lastparttype=$(echo "$parts" | sed -n -e '$ s/.*type=\([A-F0-9-]\+\).*/\1/p')
+if [ "$lastparttype" = "BFBFAFE7-A34F-448A-9A5B-6213EB736C22" ]; then
+  # We'll use the free space up to where this partition starts
+  preserve_start=$(echo "$parts" | sed -n -e '$ s/.*start=[ ]*\([^,]*\).*/\1/p')
+
+  # Save the partition entry to be restored later, and delete it from the
+  # list for now, to simplify the logic that follows.
+  preserve_partition=$(echo "$parts" | sed -n -e '$ s/[^:]*:\(.*\)$/\1/p')
+  parts=$(echo "$parts" | sed '$d')
+
+  # Recalculate lastpartno now that we've temporarily excluded the last
+  # partition
+  lastpartno=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
```

Comment:
You don't use `lastpartno` in this `if` block – could we do this initialization once, just before it's used?